src/directory/mmap_directory.rs

```diff
@@ -40,9 +40,11 @@ fn open_mmap(full_path: &Path) -> result::Result<Option<MmapReadOnly>, OpenReadE
         // instead.
         return Ok(None);
     }
-    MmapReadOnly::open(&file)
-        .map(Some)
-        .map_err(|e| From::from(IOError::with_path(full_path.to_owned(), e)))
+    unsafe {
+        MmapReadOnly::open(&file)
+            .map(Some)
+            .map_err(|e| From::from(IOError::with_path(full_path.to_owned(), e)))
+    }
```

Comment:
I don't understand why this change was required when I updated fst -> 0.3 (could be new rust stable too) - but would want someone else's eyes on it.

Reply:
You mean the `unsafe`  ? There is a comment about it in the fst crate.
It was modified in a non-backward compatible way in `0.3`. As a general rule any change from `0.x` to `0.y` may change backward compatibility.
For version > 1.x, backward compatilibility is respected as long as the major version does not change.

Reply:
Cool, since it was an unsafe change I just wanted someone else to look at it. :)

Reply:
(To be really "philosophically" correct here, we should probably make the caller `unsafe`... I think this is borderline enough to not care.)